CONTRIBUTING.md

```diff
@@ -7,12 +7,38 @@ repositories will go through.  All changes, regardless of whether they are from
 newcomers to the community or from the core team follow the exact
 same process and are given the same level of review.
 
+- [How to engage](#how-to-engage)
 - [Code of conduct](#code-of-conduct)
 - [Contributor license agreements](#contributor-license-agreements)
 - [Issues](#issues)
 - [Contributing a feature](#contributing-a-feature)
 - [Pull requests](#pull-requests)
 
+## How to engage
+
+The Istio community has a number of different communication channels. It can be hard to figure out
+where's the right place to ask a question or propose a design. It can be complicated on the receiving end too,
+trying to keep track of all the incoming input. So we'd like to encourage the following:
+
+- Initial design discussions should happen on the [istio-dev](istio-dev@googlegroups.com) mailing list. This is a great
+place to float ideas and get thoughtful feedback. It's archived for everybody to consume into the future.
+
+- If a design is substantial enough, it deserves a design document. Design documents are written with Google Docs 
+and put into our [Team Drive](https://drive.google.com/corp/drive/u/0/folders/0AIS5p3eW9BCtUk9PVA). If you add a doc
+in the team drive, drop a note in istio-dev to let people know about it.
+
+- Slack is great for quick interactions that demand immediate attention. "Help, XXX doesn't work anymore" or
+"my cluster is on fire, how do I put it out?". Slack is not really appropriate for protracted design discussions which
+usually benefit from more deliberate and thoughtful interactions possible in email.
+
+Keep in mind that Slack can be highly disruptive to the large number of users monitoring the channels. As such,
+please keep funny threads, jokes, memes, etc, out of the main channels. Try to quickly move discussions into
```

Comment:
Maybe point folks to a 'misc' for idle musings, don't want to sound too grinchy :)